src/wallet/rpcwallet.cpp

```diff
@@ -513,6 +514,18 @@ UniValue sendtoaddress(const JSONRPCRequest& request)
         }
     }
 
+    if (!request.params[8].isNull()) {
+        OutputStyle style = ParseStyle(request.params[8].get_str());
+        if (style == STYLE_NONE) {
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Unknown address style '%s'", request.params[8].get_str()));
+        }
+        // Generate a new key that is added to wallet
+        CPubKey newKey;
+        if (!pwallet->GetKeyFromPool(newKey)) {
+            throw JSONRPCError(RPC_WALLET_KEYPOOL_RAN_OUT, "Error: Keypool ran out, please call keypoolrefill first");
+        }
+        coin_control.destChange = pwallet->GetDestinationForKey(newKey, style);
```

Comment:
I think it is preferable to add `m_change_style` to coin control so that a new key is consumed only when necessary. It can also helps in other send/fund calls. This should be used in:
https://github.com/bitcoin/bitcoin/blob/f74459dba6de4d4462860318f6ee5bda8522e07b/src/wallet/wallet.cpp#L2679